substrate/client/network/src/peer_store.rs

```diff
@@ -19,7 +19,8 @@
 //! [`PeerStore`] manages peer reputations and provides connection candidates to
 //! [`crate::protocol_controller::ProtocolController`].
 
-use libp2p::PeerId;
+use crate::PeerId;
```

Comment:
The idea was that the backend can use the specific `PeerId` (libp2p/litep2p) without introducing `sc-network-types::PeerId` everywhere in `sc-network` because it would be unnecessary. The new `PeerId` is only used when there is an interface between two components, e.g., `sc-network-sync` and `sc-network`. Does that make sense?

Reply:
Not sure if I got the point. My understanding was that we use `sc_network_types::PeerId` everywhere in substrate, except when we interact with a specific networking library / backend. In this sense `PeerStore` is a generic component, so why it is not using a generic `PeerId`?

Reply:
`PeerStore` is a special case because I think it can be made generic and thus could be converted to use `sc_network_types::PeerId` once both backends use it but that is not the case now.

This type was introduced solely for the purpose of providing a generic `PeerId` where that it is needed. `Notifications` calling `ProtocolController` doesn't have to use the generic `PeerId` because both `Notifications` and `ProtocolController` are libp2p-specific concepts so there is no value, at least none that I can see, in using `sc_network_types::PeerId` when `Notification` reports to `ProtocolController` that an outbound substream was rejected. Both of them live in the libp2p portion of the code base so they can freely use `libp2p::PeerId`